lib/libc/mips/string/jdw57_memcpy.c

```diff
@@ -120,17 +119,19 @@ bcopy(const void *src0, void *dst0, size_t length)
 {
 	if (length == 0 || src0 == dst0)		/* nothing to do */
 		goto done;
-		
+
 #ifdef __CHERI__
-	char * CAPABILITY dst = __builtin_cheri_bounds_set((char * CAPABILITY)dst0,length);
-	const char * CAPABILITY src = __builtin_cheri_bounds_set((const char * CAPABILITY)src0,length);
+	char * CAPABILITY dst =
+	    __builtin_cheri_bounds_set((__cheri_cast void * CAPABILITY)dst0,length);
+	const char * CAPABILITY src =
+	    __builtin_cheri_bounds_set((__cheri_cast const void * CAPABILITY)src0,length);
 #else
-	char * dst = (char * )dst0;
-	const char * src = (const char * )src0;
+	char *dst = dst0;
+	const char *src = src0;
 #endif
 	size_t t;
-	
-#if defined(MEMMOVE) || defined(BCOPY) || defined(CMEMMOVE_C) || defined(CMEMMOVE)
```

Comment:
Why was CMEMMOVE_C removed on this line?  This shouldn't break freebsd, but is needed for using this file in the baremetal test suite.

Reply:
I removed CMEMMOVE_C here because no other instances existed (e.g. there wasn't a declaration of cmemmove_c above) and thus it didn't do anything.

Reply:
Looks like this was meant to be MEMMOVE_C, the case used in the headers.  Without this flag, MEMMOVE_C will not be able to handle overlap as it should, so I guess we should put MEMMOVE_C back in.